src/ProvidedTypes.fs

```diff
@@ -13889,6 +13961,518 @@ namespace ProviderImplementation.ProvidedTypes
                     ilg.Emit(I_castclass (transType  targetTy))
 
                 popIfEmptyExpected expectedState
+                
+            | SpecificCall <@ (<) @>(None, [t1], [a1; a2]) -> 
+                emitExpr ExpectedStackState.Value a1
+                emitExpr ExpectedStackState.Value a2
+                match t1 with 
+                | Bool | SByte | Char
+                | Double | Single
+                | Int16 | Int32 | Int64 -> ilg.Emit(I_clt)
+                | Byte
+                | UInt16 | UInt32 | UInt64 -> ilg.Emit(I_clt_un)
+                | String ->
+                    ilg.Emit(I_call(Normalcall, (convTypeToTgt typeof<System.String>).GetMethod("CompareOrdinal", [|t1; t1|]) |> transMeth, None))
+                    emitExpr ExpectedStackState.Value <@@ 0 @@>
+                    ilg.Emit(I_clt)
+                | StaticMethod "op_LessThan" [|t1; t1|] m -> 
+                    ilg.Emit(I_call(Normalcall, transMeth m, None))
+                | _ -> failwithf "Operator (<) not supported for type %s" t1.Name
+
+           
+            | SpecificCall <@ (>) @>(None, [t1], [a1; a2]) -> 

```

Comment:
In general, for performance reasons, each of these uses of `SpecificCall` should be lifted to a helper pattern, defined at the module level, e.g.

```
let (|GreaterThan|_|) = (|SpecificCall|_|) <@ (>) @>
```
This prevents the repeated creation of a quotation for `<@ (>) @>` and, IIRC, the implementation of SpecificCall does some pre-processing on the partial application.

Reply:
updated